mycore-mods/src/main/java/org/mycore/mods/merger/MCRRedundantModsGenreEventHandler.java

```diff
@@ -0,0 +1,91 @@
+package org.mycore.mods.merger;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jdom2.Element;
+
+import java.util.Objects;
+
+/**
+ * Checks for and removes redundant genres in Mods-Documents. If a genre category and
+ * the genre's child category are both present in the document, the parent genre will
+ * be removed.
+ */
+public class MCRRedundantModsGenreEventHandler extends MCRAbstractRedundantModsEventHandler {
+
+    private static final Logger LOGGER = LogManager.getLogger(MCRRedundantModsGenreEventHandler.class);
+
+    protected static final String CLASSIFICATION_ELEMENT_NAME = "genre";
+
+    @Override
+    protected String getClassificationElementName() {
+        return CLASSIFICATION_ELEMENT_NAME;
+    }
+
+    /**
+     * Returns false if the authorities of the two genre elements are the same, but
+     * the displayLabels or the types differ from each other.
+     * @param el1 the first element to be compared
+     * @param el2 the first element to be compared
+     * @return false if inconsistent
+     */
+    @Override
+    protected boolean isConsistent(Element el1, Element el2) {
+        return !hasSameAuthority(el1, el2) ||
+            (checkDisplayLabelConsistence(el1, el2) && checkTypeConsistence(el1, el2));
```

Comment:
Could just be `checkDisplayLabelConsistence(el1, el2) && checkTypeConsistence(el1, el2)`.